### PR TITLE
Trio: fix settings, factories, and helpers

### PR DESCRIPTION
**Pull request by mhchia**
The rest of the code in #365, including the settings, factories, and the helper functions.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -74,6 +74,9 @@
     "pynacl==1.3.0",
     "dataclasses>=0.7, <1;python_version<'3.7'",
     "async_generator==1.10",
+    "trio>=0.13.0",
```
Comment (ChihChengLiang) [inline setup.py]: Missing `trio_typing`

Comment (mhchia) [reply setup.py]: Could you point out where we require `trio_typing`? I'm not able to find it used in the codebase.

Comment (ChihChengLiang) [reply setup.py]: Oh, please check here. https://github.com/libp2p/py-libp2p/pull/396/files#r369460417

Comment (mhchia) [reply setup.py]: Nice catch! Fixed in https://github.com/libp2p/py-libp2p/pull/365/commits/e57d01f360de9fb101adffa686cf5aff414fc8f8.

**Comment by mhchia** [comment]
Closed by #365

**Pull request closed without merging**